index.js

```diff
@@ -26,6 +26,18 @@ module.exports = function () {
 		return str;
 	}
 
+	var isLastCharLower = false;
+	for (var i = 0; i < str.length; i++) {
+		var c = str.charAt(i);
+		if (isLastCharLower && c.toUpperCase() === c) {
+			str = str.substr(0, i) + '-' + str.substr(i);
+			isLastCharLower = false;
+			i++;
+		} else {
+			isLastCharLower = (c.toLowerCase() === c);
+		}
+	}
+
```

Comment:
All this can be replaced by this

```js
str = str.replace(/([a-z])([A-Z])/g, '$1-$2');
```

which can be added in the return statement.

Although this fixes some extra use cases, it does not fix `FBBazzy-style`. It will result in `fbbazzyStyle`.